### ISSUE TITLE
Add GitHub Workflow automation [ lints, tests, audits, dependencies, release publishing ]

### ISSUE THREAD

**Issue by yonas** [pipelight/pipelight#30]
## Features
- automated formatting: `cargo fmt`, `treefmt`
- automated tests: `cargo test`, `venom`, `hurl`
- automated security audits: `cargo auditable`, `cargo audit`, `cargo vet`, `trivy`, `trufflehog`
- automated checksum and signing: `b3sum`, `rsign`, `cosign`
- automated dependency upgrading: GitHub `dependabot`
- automated changelog generation: `git-cliff`
- automated documentation: GitHub Pages, `mdbook`, `rspress`, `typst`
- automated publishing
  - releases [ version bumping, git tagging ]
  - architectures
    - amd64
    - arm v6, v7, v8
    - risc v
  - packages [ Ubuntu / Debian, Alpine, Arch, FreeBSD, OpenBSD, NetBSD, Mac OS X, Windows ]
    - Flatpack
    - AppImage
    - Snap
    - Nix
  - GitHub Action
- automated linting: `editorconfig`, `goji`, `cocogitto`, `lychee`, `typos`, `super linter`
- automated pre-commits: `lefthook`
- automated templating: `minijinja`
- automated configuration schema and data validation: `kcl`, `rcl`

## Utilities
- [just](https://just.systems)
- [taskfile](https://taskfile.dev)
- [comtrya](https://comtrya.dev)
- [minijinja](https://github.com/mitsuhiko/minijinja)
- [kcl](https://kcl-lang.io)
- [rcl](https://rcl-lang.org)

**Comment by pipelight** [comment]
Well! That is kind of a mission i'll be happy to take on!
I suppose you are comfortable with all ot this! If you have some prefered resources or some guidelines to drop btw, that would help a lot.

**Comment by opviel** [comment]
@yonas are you suggesting that you would like to write those GH-Actions? You certainly seem to know a lot about them 😉

**Comment by yonas** [comment]
@opviel Yes I can definitely help you with them. I'll setup a test repo and report back here.

**Comment by crocuda** [comment]
Fantastic!

**Comment by opviel** [comment]
Please get back to us when you have a testing setup running, and then we can discuss the scope. For example, I don't think Flatpaks/AppImages are necessary. But automated release building on git tag for example, absolutely!
